src/database/hooks/useSaveAnonChatHook.ts

```diff
@@ -121,10 +128,110 @@ const useSaveAnonChatHook = () => {
+    } else if (!isAnonymous && channelIdAsSignedUser) {
+      checkedChannel = await ChannelListSchema.getById(localDb, channelIdAsSignedUser);
+    }
+
+    const randomMessageId = uuid();
+
+    // If channel is found, save chat as pending, then navigate to chat screen
+    let currentChatSchema: ChatSchema | null = null;
+    if (checkedChannel) {
+      currentChatSchema = await ChatSchema.generateSendingChat(
+        randomMessageId,
+        (isAnonymous ? anonProfileId : signedProfileId) || '',
+        checkedChannel?.id,
+        message,
+        [],
+        localDb,
+        'regular',
+        'pending'
+      );
+
+      await currentChatSchema?.save(localDb);
+
+      helperGoToAnonymousChat(
+        {
+          message: {
+            cid: checkedChannel?.id
+          }
+        },
+        GROUP_INFO
+      );
+    }
+
+    if (!checkedChannel) setIsLoadingSendDm(true);
+    if (isAnonymous) {
+      try {
+        const response = await sendAnonymousDMOtherProfile({
+          anon_user_info_emoji_name: anonUserInfo?.anon_user_info_emoji_name,
+          anon_user_info_emoji_code: anonUserInfo?.anon_user_info_emoji_code,
+          anon_user_info_color_name: anonUserInfo?.anon_user_info_color_name,
+          anon_user_info_color_code: anonUserInfo?.anon_user_info_color_code,
+          message,
+          user_id: targetUserId
+        });
+        eventTrack?.onBioSendDm();
+        if (checkedChannel) currentChatSchema?.updateChatSentStatus(localDb, response);
+        else await saveChatFromOtherProfile(response, 'sent', !checkedChannel, ANON_PM);
+      } catch (e) {
+        console.log('error send anon dm', e);
+        if (e?.response?.data?.status === CHANNEL_BLOCKED) {
+          const response = e?.response?.data?.data;
+          await savePendingChatFromOtherProfile(response, !checkedChannel, ANON_PM);
+        }
+      }
+    } else {
+      try {
+        const response = await sendSignedDMOtherProfile({
+          message,
+          user_id: targetUserId
+        });
+
+        eventTrack?.onBioSendDm();
+
+        const newResponse = {...response, members: response?.message?.members};
+        if (checkedChannel) currentChatSchema?.updateChatSentStatus(localDb, response);
+        else await saveChatFromOtherProfile(newResponse, 'sent', !checkedChannel, SIGNED);
+      } catch (e) {
+        console.log('error send signed dm', e);
+      }
+    }
+    setIsLoadingSendDm(false);
+  };
+
   return {
+    isLoadingSendDm,
+
     saveChatFromOtherProfile,
-    savePendingChatFromOtherProfile
+    savePendingChatFromOtherProfile,
+    sendChatFromOtherProfileV2,
+    setIsLoadingSendDm
   };
 };
 
 export default useSaveAnonChatHook;
+export type SendChatFromOtherProfileParams = {
+  isAnonymous: boolean;
+  targetUserId: string;
+  message: string;
+  anonUserInfo?: AnonUserInfo;
+  channelIdAsSignedUser?: string;
+  channelIdAsAnonUser?: string;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the function `sendChatFromOtherProfileV2`, there's a lot going on which makes it hard to understand and maintain. Consider breaking it down into smaller functions. Also, error handling could be improved. Currently, errors are just logged to the console. It would be better to handle these errors in a way that the user interface can respond to them. For example, you could use an error state variable that gets updated whenever an error occurs.

Additionally, consider using a more descriptive name for the function `sendChatFromOtherProfileV2`. The `V2` suffix doesn't provide any information about what the function does or how it differs from a potential `sendChatFromOtherProfileV1`.

Lastly, the loading state `isLoadingSendDm` is set to true only when `checkedChannel` is null. This might lead to inconsistent UI behavior if `checkedChannel` is not null but the request still takes time to complete. Consider setting `isLoadingSendDm` to true at the start of the function and then set it to false in a finally block after the try-catch blocks to ensure it is always updated regardless of whether an error occurs or not.

Here's a simplified version of how you could refactor the function:

```javascript
const sendChatFromOtherProfile = async (params: SendChatFromOtherProfileParams) => {
  setIsLoadingSendDm(true);
  try {
    const checkedChannel = await getCheckedChannel(params);
    const currentChatSchema = await prepareChatSchema(checkedChannel, params);
    await sendChatMessage(params, checkedChannel, currentChatSchema);
  } catch (error) {
    console.log('Error sending chat message:', error);
    // Update an error state variable here so the UI can respond to the error
  } finally {
    setIsLoadingSendDm(false);
  }
};
```

In this version, `getCheckedChannel`, `prepareChatSchema`, and `sendChatMessage` would be separate functions that handle specific parts of the process. This makes each function simpler and easier to understand, and it also makes it easier to handle errors in a granular way.


<!-- This is an auto-generated comment by OSS CodeRabbit -->